src/controllers/dynakube/oneagent/reconciler.go

```diff
@@ -35,22 +37,27 @@ func NewOneAgentReconciler( //nolint:revive // maximum number of return results
 	client client.Client,
 	apiReader client.Reader,
 	scheme *runtime.Scheme,
-	clusterID string) *Reconciler {
+	clusterID string,
+	dynakube *dynatracev1beta1.DynaKube) *Reconciler {
+	proxyReconciler := proxy.NewReconciler(client, apiReader, scheme, dynakube)
+
 	return &Reconciler{
-		client:    client,
-		apiReader: apiReader,
-		scheme:    scheme,
-		clusterID: clusterID,
+		client:          client,
+		apiReader:       apiReader,
+		scheme:          scheme,
+		clusterID:       clusterID,
+		proxyReconciler: proxyReconciler,
 	}
 }
 
 type Reconciler struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
-	client    client.Client
-	apiReader client.Reader
-	scheme    *runtime.Scheme
-	clusterID string
+	client          client.Client
+	apiReader       client.Reader
+	scheme          *runtime.Scheme
+	clusterID       string
+	proxyReconciler controllers.Reconciler
```

Comment:
you should do this somewhere in the [dynakube controller](https://github.com/Dynatrace/dynatrace-operator/blob/2328a307c407168f5b319c8bd278a92409bab305/src/controllers/dynakube/dynakube_controller.go#L248), because both the activegate and oneagent can need it, so it should be a more central location

Reply:
Great point, I didn't recalled about this :+1::+1::+1: